src/main/java/com/aws/greengrass/builtin/services/lifecycle/LifecycleIPCEventStreamAgent.java

```diff
@@ -352,6 +352,12 @@ public PauseComponentResponse handleRequest(PauseComponentRequest request) {
                     throw new InvalidArgumentsError("Component name is required.");
                 }
 
+                // A component should not be allowed to pause itself because it will never be resumed and can
+                // only be revived via a restart. If it needs it should just exit instead of trying to pause itself.
+                if (componentName.equals(serviceName)) {
```

Comment:
why not? This seems like something people will want...
It can pause itself and be resumed by someone else right?

Reply:
Currently no other component will know about it being paused. We can easily allow this later if we add a mechanism for knowing if a component is paused, either through a state or a new IPC API.

Reply:
What's the point of limiting this though? Customers _could_ have their own mechanism, like sending a pubsub message before pausing.

Reply:
Changed it back to how it was